wagtailgeowidget/widgets.py

```diff
@@ -39,153 +40,627 @@
+                        GOOGLE_MAPS_V3_APIKEY,
+                        GOOGLE_MAPS_V3_LANGUAGE,
+                    ),
+                ]
+
+            return forms.Media(
+                js=js,
+            )
+
+        def render(self, name, value, attrs=None, renderer=None):
+            try:
+                attrs["id"]
+            except (KeyError, TypeError):
+                raise TypeError(
+                    "GeocoderField cannot be rendered without an 'id' attribute"
+                )
+
+            value_data = value
+            widget_html = super().render(name, value_data, attrs)
+
+            return mark_safe(widget_html)
+
+    class LeafletField(forms.HiddenInput):
+        address_field = None
+        zoom_field = None
+        id_prefix = "id_"
+        srid = None
+        hide_latlng = False
+
+        def __init__(self, *args, **kwargs):
+            self.address_field = kwargs.pop("address_field", self.address_field)
+            self.zoom_field = kwargs.pop("zoom_field", self.zoom_field)
+            self.srid = kwargs.pop("srid", self.srid)
+            self.hide_latlng = kwargs.pop("hide_latlng", self.hide_latlng)
+            self.id_prefix = kwargs.pop("id_prefix", self.id_prefix)
+            self.zoom = kwargs.pop("zoom", GEO_WIDGET_ZOOM)
+
+            # Keeps a reference to the value data from the render method
+            self.value_data = None
 
-    def render_js_init(self, id_, name, value):
-        data = {
-            "defaultLocation": GEO_WIDGET_DEFAULT_LOCATION,
-            "addressField": self.address_field,
-            "zoomField": self.zoom_field,
-            "zoom": self.zoom,
-            "srid": self.srid,
-            "showEmptyLocation": GEO_WIDGET_EMPTY_LOCATION,
-            "translations": translations,
-        }
-
-        if value and isinstance(value, str):
-            result = geosgeometry_str_to_struct(value)
-            if result:
+            super().__init__(*args, **kwargs)
+
+        def build_attrs(self, *args, **kwargs):
+            data = {
+                "defaultLocation": GEO_WIDGET_DEFAULT_LOCATION,
+                "addressField": self.address_field,
+                "zoomField": self.zoom_field,
+                "zoom": self.zoom,
+                "srid": self.srid,
+                "tileLayer": GEO_WIDGET_LEAFLET_TILE_LAYER,
+                "tileLayerOptions": GEO_WIDGET_LEAFLET_TILE_LAYER_OPTIONS,
+                "showEmptyLocation": GEO_WIDGET_EMPTY_LOCATION,
+                "translations": translations,
+            }
+
+            if self.value_data and isinstance(self.value_data, str):
+                result = geosgeometry_str_to_struct(self.value_data)
+                if result:
+                    data["defaultLocation"] = {
+                        "lat": result["y"],
+                        "lng": result["x"],
+                    }
+
+            if self.value_data and isinstance(self.value_data, Point):
```

Comment:
This fails if not using GeoDjango. 
```suggestion
            if Point and self.value_data and isinstance(self.value_data, Point):
```

Reply:
@jomiq , Thanks, applied the suggestion.